### DISCUSSION TITLE
Enable Brotli compression for Responses?

### DISCUSSION THREAD

**Discussion by edent** [symfony/symfony#51074] [Q&A]
My server supports Brotli. Static HTML & JSON served from `/public/` has `content-encoding:br`.

But if I use `JsonResponse` to dynamically generate and serve JSON, it only has `gzip` compression.

I've searched the documentation - and this forum - but I can't find anything about enabling Brotli compression.  Is this something which can be done?

**Comment by Krinkle** [comment]
Do you use Apache or nginx?

What did you do to enable Brotli for static files? Is configured by file extension or mime type? I'd consider letting the server handle it transparently, which might require letting PHP respond without compression.

**Reply by edent** [reply]
I believe it is Apache. It is a managed server, so I didn't have to enable anything.

When I download `/public/test.json` it is served as Brotli compressed. When I dynamically generate a `JsonResponse` with Symfony, it is only gzip compressed.  I don't do anything to specify the compression in Symfony.

So, I *want* to let the server handle it - but Symfony doesn't seem to allow that.

**Comment by edent** [comment]
Answer is... I'm an idiot! The JsonResponse was being gzipped because that was *smaller* than the Brotli'd version.
